### PR TITLE
Add Koolertron AMAG23 Keyboard

### PR DESCRIPTION
**Pull request by ianmclinden**
## Description

Adds Koolertron AMAG23 - 23 Key market available macro pad, with RGB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/amag23/amag23.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef AMAG23_H
+#define AMAG23_H
```
Comment (drashna) [inline keyboards/amag23/amag23.h:17]:
```suggestion
#pragma once
```

File: keyboards/amag23/amag23.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    K30, K31, K32, K33,    K34    \
+) { \
+    { K00, K01, K02, K03, K04,   K05 }, \
+    { K10, K11, K12, K13, K14,   K15 }, \
+    { K20, K21, K22, K23, K24,   K25 }, \
+    { K30, K31, K32, K33, KC_NO, K34 }  \
+}
+// clang-format on
+
+#endif /* AMAG23_H */
```
Comment (drashna) [inline keyboards/amag23/amag23.h:35]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by ianmclinden** [comment]
updated to conform with requests

File: keyboards/amag23/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      KC_NO,   KC_NO,   KC_NO,   KC_NO,            KC_NO
+  ),
+};
+// clang-format on
+
+void matrix_init_user(void) {
+}
+
+void matrix_scan_user(void) {
+}
```
Comment (drashna) [inline keyboards/amag23/keymaps/default/keymap.c:42]:
```suggestion
```

File: keyboards/amag23/keymaps/qwert/config.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
```
Comment (drashna) [inline keyboards/amag23/keymaps/qwert/config.h:16]: This file could be safely removed.

File: keyboards/amag23/keymaps/qwert/keymap.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      KC_TR,   KC_TR,   KC_TR,   KC_TR,            KC_TR
+  ),
+};
+// clang-format on
+
+void matrix_init_user(void) {
+}
+
+void matrix_scan_user(void) {
+}
```
Comment (drashna) [inline keyboards/amag23/keymaps/qwert/keymap.c:44]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/amag23/keymaps/qwert/config.h (hunk 4 above)
Comment (ianmclinden) [reply keyboards/amag23/keymaps/qwert/config.h:16]: does this also apply to `keymaps/default/config.h` ?

**Review by drashna — APPROVED** [review]

File: keyboards/amag23/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define BASE 0
+#define FN1  1
```
Comment (zvecr) [inline keyboards/amag23/keymaps/default/keymap.c:19]:
```suggestion
// Defines names for use in layer keycodes and the keymap
enum layer_names {
    BASE,
    FN1
};
```

File: keyboards/amag23/keymaps/qwert/keymap.c [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#define BASE 0
+#define FN1  1
```
Comment (zvecr) [inline keyboards/amag23/keymaps/qwert/keymap.c:19]:
```suggestion
// Defines names for use in layer keycodes and the keymap
enum layer_names {
    BASE,
    FN1
};
```

File: keyboards/amag23/keymaps/qwert/keymap.c [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+#define KC_TR KC_TRNS
+
```
Comment (zvecr) [inline keyboards/amag23/keymaps/qwert/keymap.c:22]:
```suggestion

```

File: keyboards/amag23/keymaps/qwert/keymap.c [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,
+      MO(FN1), KC_A,    KC_S,    KC_D,    KC_F,    KC_G,
+      KC_LSFT, KC_Z,    KC_X,    KC_C,             KC_SPC
+  ),
+  [FN1] = LAYOUT_all( /* Function Layer */
+      KC_TR,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,
+      KC_TR,   RGB_TOG, RGB_VAI, RGB_MOD, RGB_HUI, KC_TR,
+      KC_TR,   KC_TR,   RGB_VAD, RGB_RMOD,RGB_HUD, KC_TR,
+      KC_TR,   KC_TR,   KC_TR,   KC_TR,            KC_TR
```
Comment (zvecr) [inline keyboards/amag23/keymaps/qwert/keymap.c:35]:
```suggestion
  [FN1] = LAYOUT_all( /* Function Layer */
      _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,
      _______, RGB_TOG, RGB_VAI, RGB_MOD, RGB_HUI, _______,
      _______, _______, RGB_VAD, RGB_RMOD,RGB_HUD, _______,
      _______, _______, _______, _______,          _______
```

File: keyboards/amag23/rules.mk [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # (Legacy) Enable Bluetooth with the Adafruit EZ-Key HID.
+BLUETOOTH = no              # Current options are AdafruitEzKey, AdafruitBLE, RN42
+SPLIT_KEYBOARD = no         # Enables split keyboard support
+WAIT_FOR_USB  = no          # Wait for a USB connection before starting up
+NO_USB_STARTUP_CHECK = no   # Disables usb suspend check after keyboard startup
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+
```
Comment (zvecr) [inline keyboards/amag23/rules.mk:31]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
WS2812_DRIVER = i2c
```

File: keyboards/amag23/info.json [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+              {"label":"K21", "x":1, "y":2},
+              {"label":"K22", "x":2, "y":2},
+              {"label":"K23", "x":3, "y":2},
+              {"label":"K24", "x":4, "y":2},
+              {"label":"K25", "x":5, "y":2},
+              {"label":"K30", "x":0, "y":3},
+              {"label":"K31", "x":1, "y":3},
+              {"label":"K32", "x":2, "y":3},
+              {"label":"K33", "x":3, "y":3},
+              {"label":"K34", "x":4, "y":3}
```
Comment (zvecr) [inline keyboards/amag23/info.json:32]:
```suggestion
              {"label":"K34", "x":4, "y":3, "w":2}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**